odoo/addons/base/models/ir_ui_view.py

```diff
@@ -682,17 +687,12 @@ def _get_inheriting_views(self):
         views = self.browse(row[0] for row in rows)
 
         # optimization: fill in cache of inherit_id and mode
-        self._fields['inherit_id']._insert_cache(views, [row[1] for row in rows])
-        self._fields['mode']._insert_cache(views, [row[2] for row in rows])
-
-        # During an upgrade, we can only use the views that have been
-        # fully upgraded already.
-        if self.pool._init and not self._context.get('load_all_views'):
-            views = views._filter_loaded_views()
+        for index, f in enumerate(field_names):
```

Comment:
Nitpick. You could store `ids, *columns = zip(*rows)` and iterate over `for f, column in zip(field_names, columns, strict=True)`.